src/main/java/io/localmotion/picture/sanitizer/PictureSanitizerImplementation.java

```diff
@@ -0,0 +1,109 @@
+			if((file != null) && file.exists() && file.canRead() && file.canWrite()) {
+				try(ImageInputStream imageInputStream = ImageIO.createImageInputStream(file)) {
+					Iterator<ImageReader> imageReaderIterator = ImageIO.getImageReaders(imageInputStream);
+					if ( imageReaderIterator.hasNext()) {
+						ImageReader imageReader = imageReaderIterator.next();
+						formatName = imageReader.getFormatName();
+						fallbackOnApacheCommonsImaging = false;
+					} else {
+						ImageInfo imageInfo = Imaging.getImageInfo(file);
+						if (imageInfo != null && imageInfo.getFormat() != null && imageInfo.getFormat().getName() != null) {
+							formatName = imageInfo.getFormat().getName();
+							fallbackOnApacheCommonsImaging = true;
+						} else {
+							throw new IOException("Format of the original image is not supported for read operation !");
+						}
+					}
+				}
+
+				// Load the image
+				BufferedImage originalImage;
+				if (fallbackOnApacheCommonsImaging) {
+					originalImage = Imaging.getBufferedImage(file);
+				} else {
+					originalImage = ImageIO.read(file);
+				}
+
+				// Check that image has been successfully loaded
+				if (originalImage == null) {
+					throw new IOException("Cannot load the original image !");
+				}
+
+				// Get current Width and Height of the image
+				int originalWidth = originalImage.getWidth(null);
+				int originalHeight = originalImage.getHeight(null);
+
+
+				// Resize the image by removing 1px on Width and Height
+				Image resizedImage = originalImage.getScaledInstance(200, 200, Image.SCALE_SMOOTH);
+
+				// Resize the resized image by adding 1px on Width and Height - In fact set image to is initial size
+				Image initialSizedImage = resizedImage.getScaledInstance(originalWidth, originalHeight, Image.SCALE_SMOOTH);
+
+				// Save image by overwriting the provided source file content
+				BufferedImage sanitizedImage = new BufferedImage(resizedImage.getWidth(null), resizedImage.getHeight(null), BufferedImage.TYPE_INT_RGB);
+				Graphics graphics = sanitizedImage.getGraphics();
+				graphics.drawImage(resizedImage, 0, 0, null);
+				graphics.dispose();
+
+				pictureCompressor.compress(sanitizedImage, file, formatName);
+				// Set state flag
+				safeState = true;
+			}
+		}catch (Exception e) {
+			safeState = false;
+			log.warn("Error during Image file processing !", e);
+		}
+		return safeState;
+	}
+
+	@Override
+	public String Pictureinfo(File file) {
+		String formatName = null;
+
+		if((file != null) && file.exists()) {
+			try(ImageInputStream imageInputStream = ImageIO.createImageInputStream(file)) {
+				Iterator<ImageReader> imageReaderIterator = ImageIO.getImageReaders(imageInputStream);
+				if ( imageReaderIterator.hasNext()) {
+					ImageReader imageReader = imageReaderIterator.next();
+					formatName = imageReader.getFormatName();
+				}
+			} catch (IOException ex) {
+				log.warn("Error during Image file processing !", ex);
+			}
+		}
+		return formatName;
```

Comment:
If you are just going to return the format name, do not call the pictureinfo, just call it getPictureFormat